proxystore/store/dim/ucx.py

```diff
@@ -0,0 +1,305 @@
+        store data to.
+
+        Args:
+            name (str): name of the store instance.
+            interface (str): The network interface to use
+            port (int): the desired port for the UCX server
+            cache_size (int): size of LRU cache (in # of objects). If 0,
+                the cache is disabled. The cache is local to the Python
+                process (default: 16).
+            stats (bool): collect stats on store operations (default: False).
+        """
+        global server_process
+
+        if ucx_import_error is not None:  # pragma: no cover
+            raise ucx_import_error
+
+        logger.debug('Instantiating client and server')
+
+        self.host = get_ip_address(interface)
+        self.port = port
+
+        self.addr = f'{ucp.get_address(ifname=interface)}:{self.port}'
+
+        self._loop = asyncio.new_event_loop()
+
+        if server_process is None:
+            server_process = Process(target=self._start_server)
+            server_process.start()
+
+            # allocate some time to start the server process
+            sleep(0.2)
+
+        super().__init__(
+            name,
+            cache_size=cache_size,
+            stats=stats,
+            kwargs={'interface': interface, 'port': self.port},
+        )
+
+    def _start_server(self) -> None:
+        """Launch the local UCX server (Peer) process."""
+        logger.info(
+            f'starting server on host {self.host} with port {self.port}',
+        )
+
+        # create server
+        ps = UCXServer(self.host, self.port)
+        asyncio.run(ps.launch())
+
+        logger.info(f'Server running at address {self.addr}')
+
+    def create_key(self, obj: Any) -> UCXStoreKey:
+        return UCXStoreKey(
+            ucx_key=utils.create_key(obj),
+            obj_size=len(obj),
+            peer=self.addr,
+        )
+
+    def evict(self, key: UCXStoreKey) -> None:
+        logger.debug(f'Client issuing an evict request on key {key}.')
+
+        event = serialize({'key': key.ucx_key, 'data': None, 'op': 'evict'})
+        self._loop.run_until_complete(self.handler(event, key.peer))
+
+        self._cache.evict(key)
+
+    def exists(self, key: UCXStoreKey) -> bool:
+        logger.debug(f'Client issuing an exists request on key {key}.')
+
+        event = serialize(
+            {'key': key.ucx_key, 'data': None, 'op': 'exists'},
+        )
+        return deserialize(
+            self._loop.run_until_complete(self.handler(event, key.peer)),
+        )
+
+    def get_bytes(self, key: UCXStoreKey) -> bytes | None:
+        res: bytes | None
+        logger.debug(f'Client issuing get request on key {key}.')
+
+        event = serialize({'key': key.ucx_key, 'data': '', 'op': 'get'})
+        res = self._loop.run_until_complete(self.handler(event, key.peer))
+
+        try:
+            s = deserialize(res)
+
+            if isinstance(s, Status) and not s.success:
```

Comment:
`Status` is checked here, do the other operations also need to check `Status`?

Reply:
Yes, but I don't think the other stores throw errors just yet. I'd need to add test cases in `store_*` if the dim need to raise errors, unless we ignore KeyErrors, but raise for every other error? I still need to determine the scope of the errors that can arise with ucx, margo and websockets

Reply:
Yeah, KeyErrors in this case are always safe and expected. For UXCStore it seems okay as is since the UCXServer will crash on exceptions. Websockets looks the same as UCX, but MargoServer catches all exceptions so those need to be handled somewhere.

Since we don't know the scope of errors yes we should default to raising everything. As we understand them better we can more narrowly handle/suppress errors as needed.